src/ApplicationInsights.Kubernetes/K8sQueryClient.cs

```diff
@@ -40,103 +43,102 @@ public K8sQueryClient(IKubeHttpClient kubeHttpClient)
+            string url = Invariant($"api/v1/namespaces/{KubeHttpClient.Settings.QueryNamespace}/pods/{podName}");
+
+            return GetItemAsync<K8sPod>(url, cancellationToken);
         }
         #endregion
 
         #region Replica Sets
-        public Task<IEnumerable<K8sReplicaSet>> GetReplicasAsync()
+        public Task<IEnumerable<K8sReplicaSet>> GetReplicasAsync(CancellationToken cancellationToken)
         {
             EnsureNotDisposed();
 
             string url = Invariant($"apis/apps/v1/namespaces/{KubeHttpClient.Settings.QueryNamespace}/replicasets");
-            return GetAllItemsAsync<K8sReplicaSet>(url);
+            return GetAllItemsAsync<K8sReplicaSet>(url, cancellationToken);
         }
         #endregion
 
         #region Deployment
-        public Task<IEnumerable<K8sDeployment>> GetDeploymentsAsync()
+        public Task<IEnumerable<K8sDeployment>> GetDeploymentsAsync(CancellationToken cancellationToken)
         {
             EnsureNotDisposed();
 
             string url = Invariant($"apis/apps/v1/namespaces/{KubeHttpClient.Settings.QueryNamespace}/deployments");
-            return GetAllItemsAsync<K8sDeployment>(url);
+            return GetAllItemsAsync<K8sDeployment>(url, cancellationToken);
         }
         #endregion
 
         #region Node
-        public Task<IEnumerable<K8sNode>> GetNodesAsync()
+        public Task<IEnumerable<K8sNode>> GetNodesAsync(CancellationToken cancellationToken)
         {
             EnsureNotDisposed();
 
             string url = Invariant($"api/v1/nodes");
-            return GetAllItemsAsync<K8sNode>(url);
+            return GetAllItemsAsync<K8sNode>(url, cancellationToken);
         }
         #endregion
 
-        #region ContainerStatus
-        /// <summary>
-        /// Gets the container status for the pod, where the current container is running upon.
-        /// </summary>
-        /// <returns></returns>
-        public async Task<ContainerStatus> GetMyContainerStatusAsync()
+        private async Task<IEnumerable<TEntity>> GetAllItemsAsync<TEntity>(string relativeUrl, CancellationToken cancellationToken)
         {
-            EnsureNotDisposed();
+            Uri requestUri = GetQueryUri(relativeUrl);
+            using HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, requestUri);
+            _logger.LogTrace("Default Header: {0}", KubeHttpClient.DefaultRequestHeaders);
 
-            string myContainerId = KubeHttpClient.Settings.ContainerId;
-            K8sPod myPod = await GetMyPodAsync().ConfigureAwait(false);
-            return myPod.GetContainerStatus(myContainerId);
+            using HttpResponseMessage response = await KubeHttpClient.SendAsync(request, cancellationToken).ConfigureAwait(false);
+            if (response.IsSuccessStatusCode)
+            {
+                _logger.LogTrace("Query succeeded.");
+                string resultString = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
```

Comment:
```suggestion
                string resultString = await response.Content.ReadAsStringAsync(cancellationToken).ConfigureAwait(false);
```

Reply:
Yup. Unfortunately, the overload that takes in the cancellation token is not there in Netstandard 2.0 :-)

https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpcontent.readasstringasync?view=netstandard-2.0